models/base_model.py

```diff
@@ -819,7 +849,7 @@ def compute_D_loss_generic(
 
         if fake_name is None:
             fake = getattr(self, "fake_" + domain_img + "_pool").query(
-                getattr(self, "fake_" + domain_img + noisy)
+                getattr(self, "fake_" + domain_img + "_with_context" + noisy)
```

Comment:
It's strange that the "generic" method operates on the "_with_context" fake image. I think that `fake_with_context` should not be computed if `context_pixels` is 0.

You could do something similar to what is done with "noisy"?
```python
context = ""
if self.data_context_pixels > 0:
    context = "_with_context"
```